src/main/java/cn/nukkit/entity/item/EntityItem.java

```diff
@@ -182,7 +182,7 @@ public boolean onUpdate(int currentTick) {
         boolean lavaResistant = item != null && item.isLavaResistant();
 
         int blockId = this.getLevelBlock().getId();
-        boolean insideOfHotBlock = blockId == BlockID.FIRE || blockId == BlockID.LAVA;
+        boolean insideOfHotBlock = blockId == BlockID.FIRE || blockId == BlockID.SOUL_FIRE || blockId == BlockID.LAVA || blockId == BlockID.STILL_LAVA;
```

Comment:
Instead of using this manual check consider creating an `isInsideOfLava` function inside of `Entity` that checks if one of the colliding blocks is lava, since `BlockLavaStill` inherits from `BlockLava` and a simple `instanceof BlockLava` could be used to check it.